src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/PcsVmrBackFlower.cs

```diff
@@ -142,11 +144,15 @@ await _vmrCloneManager.PrepareVmrAsync(
                 mapping,
                 remotes,
                 subscription.TargetBranch,
+                ShouldResetBranchToRemoteWhenPreparingRepo(),
                 cancellationToken);
             await targetRepo.CreateBranchAsync(targetBranch);
             targetBranchExisted = false;
         }
 
         return (targetBranchExisted, mapping, targetRepo);
     }
+
+    // During backflow, we're targeting a specific repo branch, so we should make sure we reset local branch to the remote one
+    protected override bool ShouldResetBranchToRemoteWhenPreparingRepo() => true;
```

Comment:
This should just be a property?

Reply:
Maybe we can go with a shorter name? Like `ShouldResetClones`?

Reply:
yeah shorter is nicer, fixed and made it a property